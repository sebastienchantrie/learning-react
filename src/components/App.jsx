import React from 'react';
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';
import routes from '../config/routes';

import '../styles/main.scss';

const App = () => (
  <div className="app">
    <Router>
      <header className={'app-header'}>
        <nav>
          <ul style={{ display: 'inline-block' }}>
            {routes.map((route, i) => (
              <li key={i}>
                <Link to={route.path}>{route.name}</Link>
              </li>
            ))}
          </ul>
        </nav>
      </header>
      <div className="app-body">
        <div className="main-content">
          {routes.map((route, i) => <Route key={i} {...route} />)}
        </div>
      </div>
    </Router>
  </div>
);

export default App;
