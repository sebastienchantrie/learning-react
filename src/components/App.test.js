import React from 'react';
import App from './App';

const wrapper = shallow(<App />);
describe('(Component) App', () => {

  it('match the snapshot', () => {
    const wrapper = renderer.create(<App />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should inject the Router component', () => {
    expect(wrapper.find('.app').length).toBe(1);
  });
});