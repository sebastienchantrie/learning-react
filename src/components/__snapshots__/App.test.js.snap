// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) App match the snapshot 1`] = `
<div
  className="app"
>
  <header
    className="app-header"
  >
    <nav>
      <ul
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            Accueil
          </a>
        </li>
        <li>
          <a
            href="/exercices/00"
            onClick={[Function]}
          >
            Ex00
          </a>
        </li>
        <li>
          <a
            href="/exercices/01"
            onClick={[Function]}
          >
            Ex01
          </a>
        </li>
        <li>
          <a
            href="/exercices/02"
            onClick={[Function]}
          >
            Ex02
          </a>
        </li>
        <li>
          <a
            href="/exercices/03"
            onClick={[Function]}
          >
            Ex03
          </a>
        </li>
        <li>
          <a
            href="/exercices/04"
            onClick={[Function]}
          >
            Ex04
          </a>
        </li>
        <li>
          <a
            href="/exercices/05"
            onClick={[Function]}
          >
            Ex05
          </a>
        </li>
        <li>
          <a
            href="/exercices/06"
            onClick={[Function]}
          >
            Ex06
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <div
    className="app-body"
  >
    <div
      className="main-content"
    >
      <div>
        <h1>
          React learning exercices
        </h1>
      </div>
    </div>
  </div>
</div>
`;
